packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -307,6 +307,20 @@ const launchRole = new iam.Role(this, 'LaunchRole', {
 portfolio.setLaunchRole(product, launchRole);
 ```
 
+You can also set the launch role to use a role of a given name that exists in the local account. This can be useful when sharing a portfolio with multiple accounts, as
+the launch role in the launch role constraint will be account-agnostic resulting in fewer resources needing to be created per shared account.
+A role with the given role name must exist in the account that created the launch constraint and the account of the user who launches a product with this launch constraint.
+
+```ts fixture=portfolio-product
+import * as iam from '@aws-cdk/aws-iam';
+
+const launchRole = new iam.Role(this, 'LaunchRole', {
+  assumedBy: new iam.ServicePrincipal('servicecatalog.amazonaws.com'),
+});
+
+portfolio.setLocalLaunchRoleName(product, 'LaunchRole');
+```
+
```

Comment:
I don't love this API. I don't like that you're forcing me to create the Role myself, and I have to know to make it trust the ServiceCatalog service principal.

How about we have 2 methods in the API?

1. `setLocalLaunchRoleName(IProduct, string): void` that creates the Role for you.
2. `setLocalLaunchRole(IProduct, iam.IRole): void` that uses `role.roleName`. If you want, you can even add validation that `role.roleName` was explicitly set.

What do you think?

Reply:
2 might make sense because the admin needs the role to create the constraint, but I am told that this role gets created by AWS Organizations/Control Tower during account setup. Basically, and Admin will create the portfolio in a cdk app and deploy it in the hub account, then share it with 1000s of spoke accounts. They prefer not to, and in some cases can't, touch the spoke accounts so I don't know that they would have the permissions to create the roles. Before this feature they didn't even share portfolios because it was too much work to set up all the launch role constraints evidently.

I will try to get more info in how the role is created but I am pretty sure it is also automated in a separate process

Reply:
To be clear, I'm taking about the hub account above. Not the spoke accounts.

Reply:
ah okay that makes sense!

Reply:
@skinny85 added this change in recent commit, I skipped the part of validating that the role name is set, because `role.roleName` returns a ref to the Role as a token, so unless you know a way to check the physicalName directly, that won't help us

also, i made it so `setLocalLaunchRoleName()` returns the role that it creates instead of void, so the user could access it and add permissions and stuff if they want

Reply:
The simplest way I could think of is use `defaultChild` on the passed `role`. If it's not `undefined`, cast it to `CfnRole`, and check whether the `roleName` property is non-`undefined`.

Although note that there might be some edge cases here where the `roleName` is always a Token, so even that might not work. If you want to skip checking that, I won't object.

Reply:
that worked, it is a string if its set and a token if not, so `Token.isUnresolved()` works